subiquitycore/core.py

```diff
@@ -276,13 +277,14 @@ def __init__(self, ui, opts):
         scale = float(os.environ.get('SUBIQUITY_REPLAY_TIMESCALE', "1"))
         updated = os.path.exists(os.path.join(self.state_dir, 'updating'))
         self.common = {
+            "application": self,
             "updated": updated,
             "ui": ui,
             "opts": opts,
             "signal": Signal(),
             "prober": prober,
             "loop": None,
-            "pool": futures.ThreadPoolExecutor(4),
+            "pool": futures.ThreadPoolExecutor(10),
```

Comment:
Does self-refresh require additional threads to work? Seems slightly unrelated (but I don't have much background).

Reply:
yeah this is dimly related. self-refresh means that more background things run in parallel -- getting the snap list from the snap store, checking for a snap update, sundry other things. In testing i found something was getting blocked because it was waiting for these network requests to complete, which is bogus really. (There's no real reason to run these network requests in a thread vs using something async like asyncio or twisted, but well). There's no real reason to limit the concurrency so maybe I should just bump this to 50 or something silly (assuming threads are still just created as needed, which I think they are).